providers/databricks/src/airflow/providers/databricks/utils/databricks.py

```diff
@@ -64,3 +64,19 @@ def validate_trigger_event(event: dict):
         RunState.from_json(event["run_state"])
     except Exception:
         raise AirflowException(f'Run state returned by the Trigger is incorrect: {event["run_state"]}')
+
+
+def validate_serverless_job_settings(content) -> bool:
+    """
+    Validate correctness of the serverless task submitted.
+    If the task is either spark_python_task, python_wheel_task or dbt_task and job_cluster_key or
+    existing_cluster_id is not supplied and environments is supplied, checks if environment_key is
+    also populated else raises a Value exception
+    """
+    valid_tasks = {"spark_python_task", "python_wheel_task", "dbt_task"}
+    for task in content["tasks"]:
+        if valid_tasks.intersection(task):
+            if not {"job_cluster_key", "existing_cluster_id"}.intersection(task):
```

Comment:
I think we also need to check "new_cluster" here

```suggestion
            if not {"job_cluster_key", "existing_cluster_id", "new_cluster"}.intersection(task):
```